fetch.js

```diff
@@ -204,6 +206,21 @@
 
         if (this._bodyBlob) {
           return Promise.resolve(this._bodyBlob)
+        } else if (this._bodyStream) {
+          var reader = this._bodyStream.getReader();
+          var chunks = [];
+          var pump = function(){
+            return reader.read().then(function(result){
+              if(!result.done){
+                chunks.push(result.value)
+                return pump()
```

Comment:
Is it possible to create a `while` loop rather than recursion? I guess not, because of async.